app/adapters/three_scale/api/instrumented_http_client.rb

```diff
@@ -5,6 +5,21 @@
 # Custom HTTP Client for 3scale API client with added instrumentation.,
 class ThreeScale::API::InstrumentedHttpClient < ThreeScale::API::HttpClient
 
+  def initialize(**)
+    super
+
+    if (system_provider_port = ENV['SYSTEM_PROVIDER_PORT'].presence)
```

Comment:
@jmprusi I think this is pretty decent way of overriding endpoints with local services. Instead of going through a Route use the internal Service and set `X-Forwarded-Host` and `X-Forwarded-Proto`. 
This could help us get rid of the `BACKEND_OVERRIDE` in APIcast and it could just detect it if proper ENV is set.

Reply:
`SYSTEM_PROVIDER_PORT` is an env automatically set by Openshift right?

Reply:
Yes.